src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
```

Comment:
Consider optimizing the logic to avoid double iteration over people. Here's a potential refactor:
const women = people.filter((p) => {
  const hasChildren = people.some((p1) => p1.mother === p.name);
  return withChildren ? (p.sex === 'f' && hasChildren) : p.sex === 'f';
});

By checking the condition within a single iteration, we improve both performance and readability.